src/Angular.js

```diff
@@ -1517,7 +1517,8 @@ function allowAutoBootstrap(document) {
     var link = document.createElement('a');
     link.href = src.value;
 
-    if (document.location.origin === link.origin) {
+    if (document.location.protocol === link.protocol && document.location.host === link.host &&
```

Comment:
Ah yes, I got `host` and `hostname` confused. Will update.

You could polyfill it, but I don't know what differences there may be across browsers and how they construct origin with `protocol` + `hostname` + `post` so this seems the best way to compare IMO.